cassandra/src/test/scala/geotrellis/spark/io/cassandra/bench/CassandraIndexStrategySpec.scala

```diff
@@ -0,0 +1,223 @@
+package geotrellis.spark.io.cassandra.bench
+
```

Comment:
@pomadchin ,

So ultimately I wasn't sure exactly where to add some sort of benchmarking to this PR.  I decided on this location, but the location might be moot.  I think I am going to recommend declining this PR for now... 

Results of benchmarking were not favorable unfortunately!  I was really hoping that leveraging the locality guarantees of the SFC would net us better performance out of Cassandra than simply ignoring it like we do now.  Unfortunately that does not seem to be the case!

The following output from this test case was performed with `ccm` with `vnodes` on with 3 instances running on a 4 core (8 w/ hyperthreading) xeon processor with 64GB of RAM:

```
Average write-time for READ optimized schema: 2451.9333333333334ms
Average write-time for WRITE optimized schema: 1119.6666666666667ms
STDDEV write-time for READ optimized schema: 973.7087495185041ms
STDDEV write-time for WRITE optimized schema: 183.95712060755415ms
...
Average read-time for READ optimized schema: 311.19ms
Average read-time for WRITE optimized schema: 135.7ms
STDDEV read-time for READ optimized schema: 170.76438123917995ms
STDDEV read-time for WRITE optimized schema: 23.697468219200122ms
```

The gap narrows if you run with replicationFactor = 2 as opposed to 1 by default, but the story still isn't good:

```
Average write-time for READ optimized schema: 1938.0ms
Average write-time for WRITE optimized schema: 1155.2666666666667ms
STDDEV write-time for READ optimized schema: 478.0826985644499ms
STDDEV write-time for WRITE optimized schema: 72.90584491124304ms
...
Average read-time for READ optimized schema: 262.4ms
Average read-time for WRITE optimized schema: 207.8ms
STDDEV read-time for READ optimized schema: 106.12699939223765ms
STDDEV read-time for WRITE optimized schema: 43.671729986342434ms
```

At this point I'm a little stuck... my intuition says there should be some way to leverage the locality guarantees provided by the SFC within Cassandra, but _if_ there is, I don't think this approach is going to get us there.

So I don't want you guys to lose anything that's in this PR that might be worth keeping around... but my suspicion is that the main thrust of it isn't worth merging.  I really appreciate your time and help though while I worked through this implementation and got to a place where I could start testing and getting some numbers to invalidate my changes.

Perhaps if nothing else it's worth recording that this experiment was conducted somewhere in your internal docs so you'll know in the future that the proverbial "juice isn't worth the squeeze".  Let me know if there's anything further I can do to help in that regard or if there's something else you'd like to see tried as part of this PR before abandoning it.

Kind Regards,
Andy